### PR TITLE
Fix some of TODOs on the contribution doc

### PR DESCRIPTION
**Pull request by erasche**
See #35, there were some TODOs that needed to be addressed. I just made some fairly generic changes, IUC members please comment on this and we can get them all patched in!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,3 +14,40 @@ inclusion.
...
+## What to contribute
+
+* Wrappers for new tools
+* Wrappers without tests
+* Visualization Plugins
+* New datatypes
+
+## What not to contribute
+
+* Things already wrapped and currently maintained by other users
```
Comment (jmchilton) [inline CONTRIBUTING.md]: Aren't we moving some of Bjoern's tools here - I feel like it is reasonable that the IUC should be transferred ownership of "important" tools - especially if they are abandoned for instance. Up to you guys though of course.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -14,3 +14,40 @@ inclusion.
 * Make sure you have added the necessary tests for your changes and they pass.
 * Open a [pull request](https://help.github.com/articles/using-pull-requests)
   with these changes.
+
+## What to contribute
+
+* Wrappers for new tools
+* Wrappers without tests
```
Comment (jmchilton) [inline CONTRIBUTING.md]: Wrappers without tests seems to be an odd thing to put here - did you mean to place it under what not to contribute?

I would also add `Updates for tools, enhancements to the wrappers, bug fixes, new tests`. I would expect that to be the most common type of contribution.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -14,3 +14,40 @@ inclusion.
...
+necessarily cover all uses of the program, but should ensure that it is
+generally working. The Galaxy Wiki has a
+[page](https://wiki.galaxyproject.org/Admin/Tools/WritingTests) on writing
+tests.
+
+The IUC strongly recommends testing with [planemo](https://github.com/galaxyproject/planemo/), which provides a simple command line utility for testing functionality
+
+```console
+$ planemo test --install_galaxy my_tool.xml
+```
```
Comment (jmchilton) [inline CONTRIBUTING.md]: This is exactly the self serving stuff I didn't want to add myself but am glad its here, thanks a bunch :).

File: CONTRIBUTING.md (hunk 2 above)
Comment (erasche) [inline CONTRIBUTING.md]: Oh, crap, you're right. Good catch.

Yes, good recommendations. Will add those.

File: CONTRIBUTING.md (hunk 1 above)
Comment (erasche) [inline CONTRIBUTING.md]: Yeah, I'm of the opinion that the IUC is happy to take abandoned tools under its wing and maintain them as part of our efforts but I don't know if that's the official position. I'd rather see tools here where they're going to be perpetually maintained.

Hence the use of "currently maintained by other users", we'd like to encourage authors who are abandoning useful tools to make a PR before they kill their code.

Comment (jmchilton) [inline CONTRIBUTING.md]: I had not noticed the currently maintained caveat - that makes a lot sense now - please ignore my comment :).

File: CONTRIBUTING.md (hunk 3 above)
Comment (erasche) [inline CONTRIBUTING.md]: Of course! It's the easiest to use option for testing tools, exactly what Galaxy needed.

**Comment by erasche** [comment]
@galaxyproject/iuc any comments?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -14,3 +14,53 @@ inclusion.
...
+## Requirements for Contributions
+
+Before a PR will be accepted, the IUC has some requirements on the
+submitted code (which we will be happy to help you achieve if you need the
+assistance).
+
+* Tools must contain tests (and test-data)
+* The tests must pass
+* The tools should pass linting by planemo (`planemo lint tool.xml`)
+* If there's a relevant paper for the tool, it should be cited with a [citation](https://wiki.galaxyproject.org/Admin/Tools/ToolConfigSyntax#A.3Ccitations.3E_tag_set) block
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: I would say either "in a citation block" or "with a \<citation> tag".

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -14,3 +14,53 @@ inclusion.
 * Make sure you have added the necessary tests for your changes and they pass.
 * Open a [pull request](https://help.github.com/articles/using-pull-requests)
   with these changes.
+
+## What to contribute
+
+* Wrappers for new tools
+* Visualization Plugins
+* New datatypes
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: I think that the general preference is that common datatypes should be added directly to Galaxy codebase.

File: CONTRIBUTING.md (hunk 4 above)
Comment (erasche) [inline CONTRIBUTING.md]: Ah, good suggestion. Done.

File: CONTRIBUTING.md (hunk 5 above)
Comment (erasche) [inline CONTRIBUTING.md]: Moved to not-recommended section with a note about it.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -14,3 +14,56 @@ inclusion.
...
+* Documentation improvements
+* New test cases
+
+### Abandoned Tools
+
+* The IUC is happy to adopt tools that you (the developer) are abandoning.
+* If there are tools that you find useful but seem to be abandoned and not
+  updated, you're welcome to create an issue recommending that the IUC adopt
+  that tool.
+
```
Comment (bgruening) [inline CONTRIBUTING.md]: Reminds me that we need some redirect feature in the TS.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -14,3 +14,56 @@ inclusion.
...
+## What not to contribute
+
+* Things already wrapped and currently maintained by other users
+* Wrappers without tests
+* New datatypes
+    * When possible, new datatypes should be added directly to the Galaxy
+      codebase as it's currently not trivial to deal with toolshed installed
+      datatypes in some situations.
```
Comment (bgruening) [inline CONTRIBUTING.md]: Actually this is pretty easy. The difficult part is versioning of datatypes. Not sure if we need to be exact here.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -14,3 +14,56 @@ inclusion.
...
+```
+
+## Requirements for Contributions
+
+Before a PR will be accepted, the IUC has some requirements on the
+submitted code (which we will be happy to help you achieve if you need the
+assistance).
+
+* Tools must contain tests (and test-data)
+* The tests must pass
```
Comment (bgruening) [inline CONTRIBUTING.md]: Add here again `planemo test --install_galaxy my_tool.xml`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -14,3 +14,53 @@ inclusion.
...
+* New datatypes
+* Updates for tools
+* Enhancements for tools (e.g. supporting new parameters)
+* Bug fixes
+* Documentation improvements
+* New test cases
+
+### Abandoned Tools
+
+* The IUC is happy to adopt tools that you (the developer) are abandoning.
```
Comment (nsoranzo) [inline CONTRIBUTING.md]: I don't think we want/are able to maintain every abandoned tool... We should probably verify that the tool is of "general" interest and there's somebody in the IUC with the necessary background to maintain it.

File: CONTRIBUTING.md (hunk 7 above)
Comment (erasche) [inline CONTRIBUTING.md]: Eh...fair point. I'll make it more specific.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -14,3 +14,56 @@ inclusion.
...
+```console
+$ planemo test --install_galaxy my_tool.xml
+```
+
+## Requirements for Contributions
+
+Before a PR will be accepted, the IUC has some requirements on the
+submitted code (which we will be happy to help you achieve if you need the
+assistance).
+
```
Comment (bgruening) [inline CONTRIBUTING.md]: Link to our BestPractices guide? https://wiki.galaxyproject.org/Tools/BestPractices

Comment (erasche) [inline CONTRIBUTING.md]: Aha! That's where that went. I'll link that.

File: CONTRIBUTING.md (hunk 8 above)
Comment (erasche) [inline CONTRIBUTING.md]: OK

File: CONTRIBUTING.md (hunk 6 above)
Comment (erasche) [inline CONTRIBUTING.md]: redirect feature?

File: CONTRIBUTING.md (hunk 9 above)
Comment (erasche) [inline CONTRIBUTING.md]: Fair point, I saw abandoned tools to be less about keeping them updated, and more just having them "owned" by someone who is able to fix them should they break down the line, with the possibility of a motivated community maintainer providing PRs for updates should they be required.

**Comment by nsoranzo** [comment]
+1 Thanks for the ultrafast feedback!

**Comment by erasche** [comment]
Sure thing, it affects everyone who wants to contribute to the IUC's repo, so I want us to be putting our best face forward ASAP since John raised the issue with #35

**Comment by erasche** [comment]
test this please (apparently cancelling builds makes jenkins forget to update github about build status)

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/21/
Test failed.

**Comment by erasche** [comment]
@galaxyproject/iuc ok to merge?

**Comment by bgruening** [comment]
+1 go ahead! :)

**Comment by erasche** [comment]
@bgruening thanks for the speedy response! 

(It's hard to know when we've reached consensus on an issue...I do like the bitbucket "approved" notifications, makes it easy to tell when it's safe to merge.)

**Comment by nsoranzo** [comment]
@erasche See https://github.com/isaacs/github/issues/92 (unofficial GitHub issue tracker).

**Comment by erasche** [comment]
@nsoranzo good to know we're not the only ones with this problem. I guess the IUC will have to solve it in the meantime by just being more strict with ourselves about +1ing things :(

**Pull request merged**